ironfish/src/network/peers/peerConnectionManager.ts

```diff
@@ -86,6 +86,13 @@ export class PeerConnectionManager {
       }
     }
 
+    if (this.peerManager.canCreateNewConnections()) {
+      const peer = this.peerManager.getRandomDisconnectedPeer()
+      if (peer) {
+        this.connectToEligiblePeers(peer)
```

Comment:
The `connectAttempts` variable was added above to avoid spending too much time establishing connections to new peers in one event loop. This new connection mechanism doesn't use it, but I think it probably should.